internal/controller/controllers/hypershiftagentserviceconfig_controller.go

```diff
@@ -118,6 +130,34 @@ func (hr *HypershiftAgentServiceConfigReconciler) getKubeconfigSecret(ctx contex
 	return secret, nil
 }
 
+func (hr *HypershiftAgentServiceConfigReconciler) ensureSpokeAgentInstallCRDs(ctx context.Context) error {
```

Comment:
Does this need to ensure the lists are exactly the same?
For example, if we add v1beta2, do need to delete the v1beta1 CRDs from the spoke?

We would need a "true" desired state calculation in that case where we list the CRDs on both the hub and spoke to ensure they're the same.

Reply:
Right. Probably a rare (uninteresting?) scenario, but added logic for deleting stale CRDs on spoke clusters.

Reply:
Please run this by the OLM team. They of course have experience managing the lifecycle of CRDs and may have some useful insights, especially in case this is more complex to handle than is obvious.

Reply:
Started a thread with OLM team.
btw, for reading the CRDs we can also go with the approach used by [registration-operator ](https://github.com/open-cluster-management-io/registration-operator/blob/79c6b55faef94d1a456cbfaeaa67f2d9f21dc89a/pkg/operators/clustermanager/controllers/clustermanagercontroller/clustermanager_controller.go#L416) of using "[embed](https://pkg.go.dev/embed#FS.ReadDir)" for the relevant [yamls](https://github.com/open-cluster-management-io/registration-operator/blob/dcf93fa3b9d6a55dd7f800b2af6a9e42d3efa01a/manifests/fs.go).